File: src/util/cookieUtil.js
import Cookies from 'universal-cookie'

const cookies = new Cookies();

export const getCookie = (key) => {
  return cookies.get(key);
}

export const setCookie = (key, value, expiredSecond) => {
  cookies.set(key, value, {
    path: "/",
    maxAge: 60 * expiredSecond,
    secure: true
  });
}

File: api/src/main/java/com/ag/domain/util/CookieUtil.java
package com.ag.domain.util;

import jakarta.servlet.http.Cookie;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.http.ResponseCookie;

import java.util.Arrays;

public class CookieUtil {

    public static ResponseCookie createHttpOnlyCookie(String name, String value, long maxAge) {
        return ResponseCookie.from(name, value)
                .maxAge(maxAge)
                .path("/")
                .httpOnly(true)
                .secure(true)
//                .sameSite("None")//or default value is Lax
                .build();
    }

    public static String getCookieValue(HttpServletRequest request, String cookieName) {
        Cookie[] cookies = request.getCookies();
        if (cookies != null) {
            return Arrays.stream(cookies)
                    .filter(cookie -> cookie.getName().equals(cookieName))
                    .findFirst()
                    .map(Cookie::getValue)
                    .orElse(null);
        } else {
            return null;
        }
    }


}
